### ISSUE TITLE
Add more granular scope to @let syntax

### ISSUE THREAD

**Issue by zygarios** [angular/angular#59195]
### Which @angular/* package(s) are relevant/related to the feature request?

compiler-cli

### Description

Currently, the scope for a local variable in Angular is limited to each block such as @for, @if, or @switch. Because of this, it's not possible to reuse variables in a more generic way, as I showed in the example. Initially, I would like to use @addressFormControls for multiple individual mat-form-field elements, so that in each of them I could use a nested variable with a generic name like controls. Unfortunately, I get an error because the name repeats what is defined above.

`    
    @let addressFormControls = employeeOffboardForm.controls.address.controls; 
    <div class="offboard__form-row">
      <mat-form-field subscriptSizing="dynamic">
        // it's ok to use nested variable
        @let control = addressFormControls.streetLine; 
        <mat-label>Street line</mat-label>
        <input [formControl]="addressFormControls.streetLine" matInput />
        @if (control.hasError("required")) {
          <mat-error>To pole jest wymagane</mat-error>
        }
        @if (control.hasError("maxlength")) {
          <mat-error>Maksymalna długość to 50 znaków</mat-error>
        }
      </mat-form-field>

      <mat-form-field subscriptSizing="dynamic">
        
        @let control = addressFormControls.streetLine;
        <mat-label>Postal code</mat-label>
        <input [formControl]="addressFormControls.postalCode" matInput />
        @if (control.hasError("required")) {
          <mat-error>To pole jest wymagane</mat-error>
        }
        @if (control.hasError("pattern")) {
          <mat-error>Nieprawidłowy kod pocztowy</mat-error>
        }
      </mat-form-field>
    </div>`

### Proposed solution

It would be very useful if the scope was applied to HTML tags. Currently, I am forced to give specific names for each one, which is not always convenient. However, if the reason for this is performance, I would understand, but I would appreciate it if you could explain why this particular model was adopted.

### Alternatives considered

An alternative is to leave it as it is now.

**Comment by JeanMeche** [comment]
`@let` works as designed, it is supposed to mimic the same behavior as regular javascript by being block-scoped.

**Comment by zygarios** [comment]
Ok, I understand that. It would just sometimes be useful to have, for example, an alternative to @let with its own scope.

Classic version:

`@let controls = formgroup.controls;`

Own scope:

```
@let (controls = formgroup.controls) { 

}
```

**Comment by zygarios** [comment]
Yes, that was also available without @let. But its would be nice to have possibility to avoid hacks Luke that :D

**Comment by wartab** [comment]
Unfortunately, the @let-block seems kinda hard to implement. The @let syntax is already kind of a [hack in the compiler](https://github.com/angular/angular/blob/main/packages/compiler/src/ml_parser/lexer.ts#L238) as it shortcuts the block parsing if it sees @let before checking any other `@xxx`. 

I wonder if it wouldn't just end up being just 
```
@block {
    @let xxx = ...;
}
```

as a pendant to this in TypeScript.

```typescript
{
    let xxx = ...;
}
```

Introducing block scopes at the HTML tag level would be a breaking change, so that won't happen.

**Comment by zygarios** [comment]
Ok, i understand :)

**Comment by JeanMeche** [comment]
To address scoping issues, we've introduced the aliasing in `@else if` blocks in 20.2. eg: 


```
@if(..) {

} @else if (foobar; as myVar) {

}
```
